cognite/neat/graph/transformers/_prune_graph.py

```diff
@@ -0,0 +1,115 @@
+from rdflib import RDF, Graph, Namespace, URIRef
+from rdflib.term import Node
+
+from ._base import BaseTransformer
+
+
+class ResolveValues(BaseTransformer):
+    """
+    Knowledge graph pruner and resolver. Will remove rdf triples from graph that does not have connections
+    to other nodes, and traverse graph for specified types to resolve the value in the final node and link it to
+    the source node.
+
+        Ex. ResolveValues:
+
+        Graph before pruning:
+        node(A, rdf:type(Pump)) -(predicate("vendor"))>
+                                node(B, rdf:type(TextObject)) -(predicate("value"))> Literal("CompanyX")
+
+        Graph after resolving values between nodes rd:type(Pump) and rdf:type(Disc):
+
+        node(A, rdf:type(Pump)) -(predicate("vendor"))> Literal("CompanyX")
+
+    Args:
+        source_node_type: RDF.type of source node
+        destination_node_type: RDF.type of edge Node
+        predicate_namespace: Namespace to use when resolving the predicate_namespace.value from the edge node to
+                             the value
+        delete_connecting_node: bool if the intermediate Node and Edge between source Node
+                                and resolved value should be deleted. Default is True
+    """
+
+    description: str = "Prunes the graph of specified rdf types that do not have connections to other nodes."
+
+    def __init__(
+        self,
+        source_node_type: URIRef,
+        destination_node_type: URIRef,
+        predicate_namespace: Namespace,
+        delete_connecting_node: bool = True,
+    ):
+        self.source_node_type = source_node_type
+        self.destination_node_type = destination_node_type
+        self.predicate_namespace = predicate_namespace
+        self.delete_connecting_node = delete_connecting_node
+
+    def transform(self, graph: Graph) -> None:
+        source_subjects = graph.subjects(object=self.source_node_type, predicate=RDF.type)
```

Comment:
This entire method can then instead leverage SPARQL query and you can avoid having expensive for-loops:

```
destion_nodes_to_remove : set[URIRef] = set()

for source_node, property_, destination_node, value in list(store.graph.query(
                    self._query_template.format(destination_node_type = IODD.TextObject,
                                               value_property = IODD.value))):
    
    # first remove old triple
    graph.remove((source_node, property_, destination_node))
    # then add new triple
    graph.add((source_node, property_, value))

    destion_nodes_to_remove.add(destination_node)
    

if self.delete_connecting_node:
    for node in destion_nodes_to_remove:
        graph.remove((node, None, None))

```